pkg/streams/streams.go

```diff
@@ -0,0 +1,108 @@
+package streams
+
+import (
+	"context"
+	"sync"
+
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// Stream is a stream of events
+type Stream struct {
+	// policy mask is a bitmap of policies that this stream is interested in
+	policyMask uint64
+	// events is a channel that is used to receive events from the stream
+	events chan trace.Event
+}
+
+// ReceiveEvents returns a channel that can be used to receive events from the stream
+func (s *Stream) ReceiveEvents() <-chan trace.Event {
+	return s.events
+}
+
+// Publish publishes an event to the stream,
+// but first check if this stream is interested in this event,
+// by checking the event's policy mask against the stream's policy mask.
+func (s *Stream) publish(ctx context.Context, event trace.Event) {
+	if s.shouldIgnorePolicy(event) {
+		return
+	}
+
+	// currently we are blocking if the channel is full,
+	// though I think we should instead drop events, because if you have
+	// two streams, one should not stop the other from receiving events,
+	// if they are not fast enough. But there also use cases where blocking
+	// might be favor, so th idea is to later make this configurable.
+	// TODO: allow this to be configurable (drop/block) (josedonizetti)
```

Comment:
awesome drop/block idea.

Reply:
Like we've spoken on teams, I believe each stream should have its own limited (size set by user ?) circular buffer and the events should come from this circular buffer. This would allow you to drop published events in the tail of the buffer if they were not consumed fast enough (which can happen depending on the network throughput/bandwidth and how fast clients consume the streams).

Reply:
Agreed, I'll create an item on the API server task to improve it later.

Reply:
As we spoke on the call, it will be implemented in a future pr.